plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/sso/bearer/BearerTokenProvider.kt

```diff
@@ -242,6 +248,29 @@ val ssoOidcClientConfigurationBuilder: (ClientOverrideConfiguration.Builder) ->
 
     // Update the RetryPolicy in the configuration
     configuration.retryPolicy(updatedRetryPolicy)
+
+    configuration.addExecutionInterceptor(object : ExecutionInterceptor {
+        override fun modifyException(context: Context.FailedExecution, executionAttributes: ExecutionAttributes): Throwable {
+            val exception = context.exception()
+            if (exception !is AwsServiceException) {
+                return exception
+            }
+
+            try {
+                val clazz = exception::class.java
+                val errorDescription = clazz.methods.firstOrNull { it.name == "errorDescription" }?.invoke(exception) as? String
```

Comment:
seems worth a comment
```suggestion
                // OIDC service has "errorDescription" field, but the SDK does not supply this information in the default exception message.
                val errorDescription = clazz.methods.firstOrNull { it.name == "errorDescription" }?.invoke(exception) as? String
```